python/cudf/cudf/core/column/column.py

```diff
@@ -1840,10 +1840,12 @@ def as_column(
         else:
             pyarrow_array = pa.array(arbitrary, from_pandas=nan_as_null)
             if isinstance(pyarrow_array.type, pa.Decimal128Type):
-                dtype = cudf.Decimal64Dtype.from_arrow(pyarrow_array.type)
+                pyarrow_dtype = cudf.Decimal64Dtype.from_arrow(
+                    pyarrow_array.type
+                )
             else:
-                dtype = arbitrary.dtype
-            data = as_column(pyarrow_array, dtype=dtype)
+                pyarrow_dtype = arbitrary.dtype
+            data = as_column(pyarrow_array, dtype=pyarrow_dtype)
```

Comment:
👍 

While we're at it, let's use `pyarrow_type` for clarity. As PyArrow calls them _types_, not _dtypes_ :(